src/DsaLocal.cc

```diff
@@ -1084,6 +1087,46 @@ void LocalAnalysis::runOnFunction(Function &F, Graph &g) {
   );
   // clang-format on
 
+  /// Mark incompelete nodes after the pass
+  /// based on the code in remove_dead
+  /// TODO: doing it inside remove_dead maybe more efficient
+  std::set<const Node *> reachable;
+  // --- collect all nodes referenced by formal parameters
+  for (auto &kv : llvm::make_range(g.formal_begin(), g.formal_end())) {
+    const Cell *C = kv.second.get();
+    if (C->isNull())
+      continue;
+    if (reachable.insert(C->getNode()).second) {
+      LOG("dsa-callsite", errs() << "\treachable formal parameter nodes " << C->getNode() << "\n";);
+      C->getNode()->setIncomplete(true);
+    }
+  }
+
+  // --- collect all nodes referenced by return parameters
+  for (auto &kv : llvm::make_range(g.return_begin(), g.return_end())) {
+    const Cell *C = kv.second.get();
```

Comment:
Separate question is whether we care about returns. If something might be passed to our procedure, that might affect how a call is resolved. If we return some memory, it is not that easy to affect what happens to the procedure.